### PR TITLE
Add tooling for checking and formatting notebooks

### PR DESCRIPTION
**Pull request by adamnsch**
Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

**Comment by adamnsch** [comment]
CI fails since docker image needs to be updated with `nbconvert`

File: requirements/notebook-ci.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 neo4j >= 5.0, < 6.0
 pyarrow >= 10.0, < 11.0
 pytest
-nbconvert == 7.2.8
+nbconvert == 7.2.9
```
Comment (FlorentinD) [inline requirements/notebook-ci.txt:4]: i would have hoped dependabot catches this. Likely the bot is not working at the moment

File: scripts/checkstyle [hunk 2]
```diff
@@ -3,9 +3,27 @@
...
+  NB=$(cat $f)
+  FORMATTED_NB=$(jupyter nbconvert \
+    --clear-output \
+    --stdout \
+    --ClearOutputPreprocessor.enabled=True \
+    --ClearMetadataPreprocessor.enabled=True \
+    --log-level CRITICAL \
+    $f)
+
+  if [[ "$FORMATTED_NB" != "$NB" ]];
```
Comment (FlorentinD) [inline scripts/checkstyle:24]: Nice! I would really like to add a check for the notebook to docs here as well.

**Review by FlorentinD — APPROVED** [review]
Great tooling!

File: requirements/notebook-ci.txt (hunk 1 above)
Comment (adamnsch) [reply requirements/notebook-ci.txt:4]: Should it always notice us on any patch release? Or just important releases with security fixes?

Comment (FlorentinD) [reply requirements/notebook-ci.txt:4]: sounds good to create a card

Comment (FlorentinD) [reply requirements/notebook-ci.txt:4]: I might have found the problem. Maybe the directory spec was wrong

Comment (adamnsch) [reply requirements/notebook-ci.txt:4]: ah ok :+1: won't add a card then

**Comment by adamnsch** [comment]
> > CI fails since docker image needs to be updated with `nbconvert`
> 
> this is still needed right?

I've updated the docker image now, so the tests should pass :) will make a PR with the change to the dockerfile too

**Pull request merged**